lib/worker/api.ts

```diff
@@ -0,0 +1,750 @@
+        }
+
+        const keyStoreID = this.keyStore.add(decryptedKey, _customIdx);
+
+        return getPrivateKeyReference(decryptedKey, keyStoreID);
+    }
+
+    async importPublicKey({ armoredKey, binaryKey }: WorkerPublicKeyImport, _customIdx?: number) {
+        const publicKey = await getKey(binaryKey || armoredKey!);
+        const keyStoreID = this.keyStore.add(publicKey, _customIdx);
+
+        return getPublicKeyReference(publicKey, keyStoreID);
+    }
+
+    async exportPublicKey<F extends SerialisedOutputFormat = 'armored'>({
+        format = 'armored',
+        keyReference
+    }: {
+        keyReference: KeyReference;
+        format?: F;
+    }): Promise<SerialisedOutputTypeFromFormat<F>> {
+        const maybePrivateKey = this.keyStore.get(keyReference._idx);
+        const publicKey = maybePrivateKey.isPrivate() ? maybePrivateKey.toPublic() : maybePrivateKey;
+        const serializedKey = format === 'binary' ? publicKey.write() : publicKey.armor();
+        return serializedKey as SerialisedOutputTypeFromFormat<F>;
+    }
+
+    /**
+     * Get the serialized private key, encrypted with the given `passphrase`.
+     * Exporting a key does not invalidate the corresponding `keyReference`, nor does it remove the key from internal storage (use `clearKey()` for that).
+     * @param options.passphrase - passphrase to encrypt the key with (non-empty string), or `null` to export an unencrypted key (not recommended).
+     * @param options.format - `'binary'` or `'armored'` format of serialized key
+     * @returns serialized encrypted key
+     */
+    async exportPrivateKey<F extends SerialisedOutputFormat = 'armored'>({
+        format = 'armored',
+        ...options
+    }: {
+        keyReference: PrivateKeyReference;
+        passphrase: string | null;
+        format?: F;
+    }): Promise<SerialisedOutputTypeFromFormat<F>> {
+        const { keyReference, passphrase } = options;
+        const privateKey = this.keyStore.get(keyReference._idx) as PrivateKey;
+        const doNotEncrypt = passphrase === null;
+        const maybeEncryptedKey = doNotEncrypt ? privateKey : await encryptKey({ privateKey, passphrase });
+
+        const serializedKey = format === 'binary' ? maybeEncryptedKey.write() : maybeEncryptedKey.armor();
+        return serializedKey as SerialisedOutputTypeFromFormat<F>;
+    }
+};
+
+export class Api extends KeyManagementApi {
+    /**
+     * Each instance keeps a dedicated key storage.
+     * @param openpgpConfig - Config entries to set as OpenPGP global configuration.
+     *                        Note: functions whose behaviour depends on it accept a `config` input option, that can be used to
+     *                        apply a configuration change limited to the single function call.
+     * @throws {Error} if the given configuration is invalid
+     */
+    constructor(openpgpConfig: OpenPGPConfig = {}) {
+        super();
+        setStandardOpenPGPConfig();
```

Comment:
Since the config is shared across the code in a worker/import (as they all use the same openpgp instance), in principle it makes more sense to set the custom config as part of `pmcrypto.init(openpgpCustomConfig)`. This would work for when the caller creates API instances directly (and not wrapped by a worker, like WorkerPool does).
However, `WorkerPool.init` then has no way to set the config for each worker (since it cannot access the corresponding pmcrypto instance).

Reply:
We are removing the option to set a custom global config, so this will be removed as well.